File: solutions/elixir/resistor-color/1/lib/resistor_color.ex
defmodule ResistorColor do
  @moduledoc false

  @spec colors() :: list(String.t())
  def colors,
    do: [
      "black",
      "brown",
      "red",
      "orange",
      "yellow",
      "green",
      "blue",
      "violet",
      "grey",
      "white"
    ]

  @spec code(String.t()) :: integer()
  def code(color),
    do: Enum.find_index(colors, &(&1 === color))
end

File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
import java.util.Map;

class ResistorColor {
	private static final Map<String, Integer> valueByResistorColors = Map.of(
			"black", 0,
			"brown", 1,
			"red", 2,
			"orange", 3,
			"yellow", 4,
			"green", 5,
			"blue", 6,
			"violet", 7,
			"grey", 8,
			"white", 9
	);

    int colorCode(String color) {
        return valueByResistorColors.get(color);
    }

	String[] colors() {
		return valueByResistorColors.entrySet()
				       .stream()
				       .sorted(Map.Entry.comparingByValue())
				       .map(Map.Entry::getKey)
				       .toArray(String[]::new);
	}
}
